azure/services/aso/aso.go

```diff
@@ -170,12 +175,14 @@ func (s *Service) CreateOrUpdateResource(ctx context.Context, spec azure.ASOReso
 		annotations[ReconcilePolicyAnnotation] = ReconcilePolicyManage
 	}
 
-	if len(labels) > 0 {
-		parameters.SetLabels(maps.Merge(parameters.GetLabels(), labels))
+	if len(labels) == 0 {
+		labels = nil
```

Comment:
Why do we want to set `labels` and `annotations` back to nil? Would it make sense instead to call `SetLabels()` and `SetAnnotations()` only when `existing == nil`?

Reply:
> Why do we want to set `labels` and `annotations` back to nil?

Because labels and annotations are both serialized with `omitempty`, the `existing` resource will never contain empty, non-nil values for either of those fields. This preemptively emulates `omitempty` for the values of `parameters` so when we calculate the diff just below these changes, we don't treat a non-nil and empty map as different from a nil one since they would both be serialized as nil.

> Would it make sense instead to call `SetLabels()` and `SetAnnotations()` only when `existing == nil`?

Any update (when `existing != nil`) may require labels and/or annotations to be updated (like when we notice a resource needs to be adopted), so I don't think that would work in all cases.